exla/lib/exla/backend.ex

```diff
@@ -67,9 +67,35 @@ defmodule EXLA.Backend do
   end
 
   @impl true
-  def to_batched_list(_out, _tensor, _backend_options) do
-    # TODO: once implemented, enable the corresponding Nx doctests in EXLA.BackendTest
-    raise "to_batched_list is not currently supported by EXLA.Backend, please call Nx.backend_copy/1"
+  def to_batched_list(out, tensor, opts) do
+    leftover = opts[:leftover]
+
+    batch_size = elem(out.shape, 0)
+    axis_size = elem(tensor.shape, 0)
+
+    remainder = rem(axis_size, batch_size)
+    num_full_batches = div(axis_size, batch_size)
+
+    to_add = if remainder != 0, do: batch_size - remainder, else: 0
+
+    {num_batches, expr_fun} =
+      if to_add != 0 and leftover == :repeat do
+        {num_full_batches + 1,
+         fn tensor, start_idx ->
+           wrapped_tensor = Nx.concatenate([tensor, Nx.slice_along_axis(tensor, 0, to_add)])
```

Comment:
We only need to do the concatenation in the last operation. So I recommend jitting the same for all slices, except the last! Or writing conditional code inside defn.

Reply:
Oh, we would do this on every call, not just once, good point!